Modules/Module_Processing/Module_ASL/xASL_wrp_RealignASL.m

```diff
@@ -506,30 +506,22 @@ function xASL_wrp_RealignASL(x, bASL)
         TempIm = xASL_io_Nifti2Im(InputPath);
         
         % Remove spikes
-        NewIm = TempIm(:,:,:, ~exclusion);
-        
         % skip this for fMRI, which is more complicated
         % due to tissue T1 effects (incomplete saturation, so temporal relation between volumes)
-        % Save in single precision, since conversion back to INT16 would
-        % otherwise loose precious precision
-        % PM: can this be simplified to same format as original??
-        xASL_io_SaveNifti(x.P.Path_ASL4D, x.P.Path_despiked_ASL4D, NewIm, 32, 0);
+        NewIm = TempIm(:,:,:, ~exclusion);
         
         % Do same for *.mat motion sidecars
         LoadParms = load(x.P.Path_ASL4D_mat, '-mat');
         mat = LoadParms.mat;
-        
         mat = mat(:,:, ~exclusion);
-
         save(x.P.Path_despiked_ASL4D_mat, 'mat');
 
         % Also change parameter vectors
-        x.Q.EchoTime = x.Q.EchoTime(~exclusion);
-        x.Q.Initial_PLD = x.Q.Initial_PLD(~exclusion);
-        x.Q.LabelingDuration = x.Q.LabelingDuration(~exclusion);
-
-        % #997 Need to save this in json sidecar for further usage in the pipeline
+        jsonFields.EchoTime = x.Q.EchoTime(~exclusion);
+        jsonFields.Initial_PLD = x.Q.Initial_PLD(~exclusion);
+        jsonFields.LabelingDuration = x.Q.LabelingDuration(~exclusion);
 
+        xASL_io_SaveNifti(x.P.Path_ASL4D, x.P.Path_despiked_ASL4D, NewIm, 32, 0, [], 1, xASL_bids_parms2BIDS(jsonFields, [], 1));

```

Comment:
the conversion parms2bids is all good. But in SaveNifti, do we really want to set the parameter bCopyOrigJson == true. I though we only want to save the three new fields without copying the entire JSON.

Reply:
Done in 9f7dc90711ec73ab569ce02e709c319e5897e2da